Update Dockerfiles and fix regression bugs

pedro-zawadniak-adi opened the pull request:
This PR fixes the regression problems by pinning the project's dependencies to tested versions. On doing so, the following changes were made:
- Creation of a west.yml file. This replaces the dependency definitions described in other repositories.
- Creation of a Dockerfile for the purpose of using the extension (as opposed to developing the extension). The development Dockerfile has been preserved. Both can be launched as Dev Containers. Upon trying to open the project as a Dev Container, the user is prompted to select which container to use.

carolinemaymaguirecmm commented on README.md:
```diff
@@ -55,11 +55,9 @@ In order to install it, go to Extensions, click `Install from VSIX...` and choos
-The Dev Container definition can be found in [.devcontainer/automl/devcontainer.json](https://github.com/antmicro/kenning-zephyr-runtime-example-app/blob/main/.devcontainer/automl/devcontainer.json) (VSCode Dev Container configuration).
-
-To start the Dev Container, click the `Reopen in Container` button from the pop-up or run the `DevContainers: Reopen in Container` action.
+To start the Dev Container, click the `Reopen in Container` button from the pop-up or run the `DevContainers: Reopen in Container` action. You'll be prompted to select which container you want to open. For general use, select "AutoML".
```
```suggestion
To start the Dev Container, click the `Reopen in Container` button in VS Code when the pop-up appears, or run the `DevContainers: Reopen in Container` command from the Command Palette. You'll then be prompted to select which container you want to open. For general use, select "AutoML".

carolinemaymaguirecmm commented on README.md:
```diff
@@ -55,11 +55,9 @@ In order to install it, go to Extensions, click `Install from VSIX...` and choos
 The easiest way to set up the environment for application and AutoML development is to develop inside a container using [Visual Studio Code Dev Containers](https://code.visualstudio.com/docs/devcontainers/containers).
 
-Sample Docker image definition can be found in [Example project using Kenning Zephyr Runtime](https://github.com/antmicro/kenning-zephyr-runtime-example-app), under [environments/Dockerfile](https://github.com/antmicro/kenning-zephyr-runtime-example-app/blob/main/environments/Dockerfile).
+This repository provides Dev Container definitions for general use and for plugin development. The Dev Container for general use is named `AutoML`, and is defined in `.devcontainer/automl/devcontainer.json`. The Dev Container for development is named `automl-embedded development` and is defined in `.devcontainer/development/devcontainer.json`.
```
```suggestion
This repository provides two Dev Container definitions: one for general use and one for developing the AutoML VS Code extension. 

- The Dev Container for general use is named `AutoML`, and is defined in `.devcontainer/automl/devcontainer.json`.  
- The Dev Container for extension development is named `automl-embedded development`, and is defined in `.devcontainer/development/devcontainer.json`.

Kemppai reviewed (approved):
Does the user guide need any updates related to this?
Approved

pedro-zawadniak-adi committed:
Add generated folders to .gitignore

This commit adds the folders generated by running the AutoML
procedure to the .gitignore file.

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
Rename Dev Container for VSCode extension development

This commit changes the Dev Container name to "automl-embedded extension development"
and moves it to the .devcontainer/development folder, along with formatting changes.

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
Create Dev Container suitable for using the automl-embedded extension

This commit creates a new Dev Container definition and and associated
Dockerfile and west.yml file. The Dev Container is named "AutoML"
and should be selected for regular use of the AutoML extension.

The purpose of this Dev Container is that it will be used instead of
relying on the Dev Container of a third-party repository.

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
Add the Dev Containers extension as a repository recommendation

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
README.md: Update to reflect recent changes

This commit updates README.md to reflect the changes introduced by
the previous commits associated with the same pull request.

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
README.md: Remove instructions for local setup

This use-case is not supported.

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
README.md: Separate instructions for AutoML plugin usage and for running the model

This repository's Dev Container can be used for running AutoML, and
antmicro/kenning-zephyr-runtime-example-app can be used to build an application
which runs the generated model.

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
README.md: Add more details to the "UART Path" setting description

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

pedro-zawadniak-adi committed:
README.md: Add section on MSDK installation

Signed-off-by: Pedro Zawadniak <Pedro.FontouraZawadniak@analog.com>

carolinemaymaguirecmm commented on README.md:
```diff
@@ -55,34 +55,20 @@ In order to install it, go to Extensions, click `Install from VSIX...` and choos
-
-**NOTE:** Kenning supports Python 3.10 or 3.11.
-In case of different Python versions available in the system, use e.g. [pyenv](https://github.com/pyenv/pyenv) to create an environment with Python 3.11 or use additional options with `pipx` installation: `--python 3.11 --fetch-missing-python`.
+MaximSDK is required to perform training on Analog Device's physical boards, such as MAX32690evkit and MAX78002evkit, and is not installed in the Dev Container by default. The Dockerfile may be modified by the user so that it performs the MaximSDK installation and sets the required environment variables. Refer to [Docker-based installation](docs/user-guide/installation/docker-install.md). Install ...
```
```suggestion
The MSDK is required to perform training on Analog Devices physical boards, such as the MAX32690 EV kit and MAX78002 EV kit, and is not installed in the Dev Container by default. The Dockerfile may be modified by the user to install the MSDK and set the required environment variables. For details, refer to [Docker-based installation](docs/user-guide/installation/docker-install.md). Installation and use are subject to acceptance of Analog Devices license terms.

carolinemaymaguirecmm commented on README.md:
```diff
@@ -55,34 +55,20 @@ In order to install it, go to Extensions, click `Install from VSIX...` and choos
 This will automatically build the image defined by Dockerfile and reopen the working directory inside the container, including all necessary software for the plugin.
 
-### Setting up a local environment
+#### Installing MaximSDK
```
```suggestion
#### Installing MSDK

carolinemaymaguirecmm commented on README.md:
```diff
@@ -143,10 +127,10 @@ Assuming the Dev Container Extension is installed in VSCode, the previously ment
 Once the Dev Container is ready, install the VSCode Extension in the Dev Container.
 
-After plugin installation, open Terminal in VSCode (it will use the environment from the Dev Container) and run the following commands in the root project directory:
+After plugin installation, open Terminal in VSCode (it will use the environment from the Dev Container) and run the following commands in the automl-embedded project directory, which is openend by default:
```
```suggestion
After installing the plugin, open a Terminal in VS Code. The terminal uses the Dev Container environment. Run the following commands in the automl-embedded project directory, which is opened by default:

carolinemaymaguirecmm commented on README.md:
```diff
@@ -158,15 +142,15 @@ The fetched Kenning Zephyr Runtime repository will be used by the plugin to buil
   Check [pyrenode3 project](https://github.com/antmicro/pyrenode3) for available variables and options.
   Renode packages are available at [builds.renode.io](https://builds.renode.io).
-* **UART Path** (optional) - path to the board's UART used for communication between Kenning and Kenning Zephyr Runtime.
+* **UART Path** (optional) - path to the board's UART used for communication between Kenning and Kenning Zephyr Runtime (e.g. `/dev/ttyUSB0`). It is recommended to set this field to the corresponding device, as it's used as a fall-back mechanism when automatic detection fails. Make sure that the device has been forwarded to the container (e.g. `ls /dev/` lists your device's name). See [Docker-bas ...
```
```suggestion
* **UART Path** (optional) - path to the board's UART used for communication between Kenning and Kenning Zephyr Runtime (e.g. `/dev/ttyUSB0`). It is recommended to set this field to the corresponding device, as it's used as a fall-back mechanism when automatic detection fails. Ensure that the device has been forwarded to the container (e.g. verify that `ls /dev/` lists your device's name). See [Docker-based installation](docs/user-guide/installation/docker-install.md) for details on using the `runArgs` setting to forward devices to Dev Containers.
